scripts/hid_configurator/configurator_core.py

```diff
@@ -639,5 +646,46 @@ def get_dfu_image_version(dfu_image):
     return ver
 
 
+class Step:
+    def __init__(self, r, g, b, substep_count, substep_time):
+        self.r = r
+        self.g = g
+        self.b = b
+        self.substep_count = substep_count
+        self.substep_time = substep_time
+
+
+def led_send_single_step(dev, recipient, step, led_id):
+    event_id = (EVENT_GROUP_LED_STREAM << GROUP_FIELD_POS) \
+               | (LED_STREAM_DATA << TYPE_FIELD_POS)
+
+    event_data = struct.pack('<BBBHHB', step.r, step.g, step.b,
+                             step.substep_count, step.substep_time, led_id)
+
+    success = exchange_feature_report(dev, recipient, event_id,
+                                      event_data, False)
+
+    return success
+
+
+def fetch_free_steps_buffer_info(dev, recipient, led_id):
+    event_id = (EVENT_GROUP_LED_STREAM << GROUP_FIELD_POS) \
+               | (led_id << MOD_FIELD_POS)
+
+    success, fetched_data = exchange_feature_report(dev, recipient,
+                                                    event_id, None, True)
+
+    if (not success) or (fetched_data is None):
+        return False, (None, None)
+
+    fmt = '<B?'
```

Comment:
``<B?`` magic string

Reply:
make it 'constnat' or just comment it.

Reply:
I've verified the documentation and sequence of charactes has meaning. The one here is a description of the packed data that means: little endian, unsigned int value, bool value. I suggest to provide such information for newcomers.